AppCenter/AppCenter/Internals/Context/UserId/MSUserIdContext.m

```diff
@@ -61,6 +62,7 @@ - (instancetype)init {
      * Center start and setUserId call.
      */
     [MS_USER_DEFAULTS setObject:[NSKeyedArchiver archivedDataWithRootObject:self.userIdHistory] forKey:kMSUserIdHistoryKey];
+    _delegates = [NSHashTable new];
```

Comment:
I think the reason we use `NSHashTable<id<MSUserIdContextDelegate>> *delegates` is to hold delegates with weak references, so you want to do it like this:

```suggestion
    _delegates = [NSHashTable weakObjectsHashTable];
```